blaze/expr/strings.py

```diff
@@ -33,6 +37,35 @@ def like(child, **kwargs):
     return Like(child, tuple(kwargs.items()))
 like.__doc__ = Like.__doc__
 
+
+class UnaryStringFunction(Expr):
+
+    """String function that only takes a single argument.
+    """
+    __slots__ = '_hash', '_child'
+
+    @property
+    def dshape(self):
+        if iscollection(self._child.dshape):
+            return datashape.var * self._dtype
+        return self._dtype
```

Comment:
return DataShape(*[self.shape + [self._dtype]])

Or alternatively define this as a subclass of `Elemwise` and rely on its `dshape` use.  This will also make these operations broadcast nicely with the chunking backends.